lms/resources/_js_config/__init__.py

```diff
@@ -41,7 +41,7 @@ def _h_user(self):
     def _application_instance(self):
         return self._lti_user.application_instance
 
-    def add_document_url(self, document_url) -> None:
+    def add_document_url(self, document_url) -> None:  # pylint: disable=too-complex
```

Comment:
This method is indeed too complex, or at least too long.

The naive approach to maybe reduce duplication here and remove some if statements is IMO not the right one.

The right solution is to avoid having one endpoint to proxy files per LMS and try to only have one. Then most of this code would disappear. Added a draft to the board about this but that would be a an ambitions reflector.